### PR TITLE
Stlx 13003 update sdk approval

### PR DESCRIPTION
**Pull request by ipelaez-simtlix**
Update sdk to recreate txRequest on pending approval

ticket : https://bitgoinc.atlassian.net/browse/STLX-13003

### CODE REVIEW

**Review by joey-bai — COMMENTED** [review]
overall lgtm, lets add more unit tests

File: modules/core/test/v2/unit/internal/tssUtils.ts
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/internal/tssUtils.ts:488]: Do we have failure cases?

Comment (ipelaez-simtlix) [reply modules/core/test/v2/unit/internal/tssUtils.ts:488]: image

It calls this endpoint, I think there is not much to test from core

File: modules/core/src/v2/pendingApproval.ts
Comment (joey-bai) [inline modules/core/src/v2/pendingApproval.ts:323]: i was testing this locally. this needs to return something like:
```
{
  txHex: txRequest.unsignedTxs[0].serializedTxHex
}
```

File: modules/core/src/v2/internal/tssUtils.ts
Comment (joey-bai) [inline modules/core/src/v2/internal/tssUtils.ts:607]: change return type to `Promise<TxRequest>` and return `txRequestSigned`

we can't use `sendTxRequest` as the final call to approve the transaction.
we need to use the existing update pending approval call instead.

if we do this, we'll create a new pending approval entry.

once you make these changes, the `sendApproval` API call will take care of broadcasting the transaction

File: modules/core/src/v2/pendingApproval.ts
Comment (joey-bai) [inline modules/core/src/v2/pendingApproval.ts:338]: you want to pass in the user's decrypted prv instead of wallet passphrase:

File: modules/core/src/v2/internal/tssUtils.ts
Comment (ipelaez-simtlix) [reply modules/core/src/v2/internal/tssUtils.ts]: let me know if now is right.

File: modules/core/src/v2/pendingApproval.ts
Comment (ipelaez-simtlix) [reply modules/core/src/v2/pendingApproval.ts]: discussed on chat, thank you for the help.

File: modules/core/src/v2/pendingApproval.ts
Comment (joey-bai) [inline modules/core/src/v2/pendingApproval.ts:65]: move this out from `ApproveOptions` and into `PendingApprovalData` - wallet platform will return this info if the pending approval is associated to a tx request

File: modules/core/src/v2/pendingApproval.ts
Comment (joey-bai) [inline modules/core/src/v2/pendingApproval.ts:324]: grab `txRequestId` from `this._pendingApproval.txRequestId`

**Review by joey-bai — APPROVED** [review]

**Review by joey-bai — APPROVED** [review]

**Pull request merged**
